scripts/smb2-capabilities.nse

```diff
@@ -0,0 +1,117 @@
+local smb = require "smb"
+local smb2 = require "smb2"
+local stdnse = require "stdnse"
+local string = require "string"
+local bit = require "bit"
+local table = require "table"
+local nmap = require "nmap"
+
+description = [[
+Attempts to list the supported capabilities in a SMBv2 server for each
+ enabled dialect.
+
+The script sends a SMB2_COM_NEGOTIATE command and parses the response
+ using the SMB dialects:
+* 2.02
+* 2.10
+* 3.00
+* 3.02
+* 3.11
+
+References:
+* https://msdn.microsoft.com/en-us/library/cc246561.aspx
+]]
+
+---
+-- @usage nmap -p 445 --script smb2-capabilities <target>
+-- @usage nmap -p 139 --script smb2-capabilities <target>
+--
+-- @output
+-- | smb2-capabilities: 
+-- |   2.02: 
+-- |     Distributed File System
+-- |   2.10: 
+-- |     Distributed File System
+-- |     Leasing
+-- |     Multi-credit operations
+--
+-- @xmloutput
+-- <table key="2.02">
+-- <elem>Distributed File System</elem>
+-- </table>
+-- <table key="2.10">
+-- <elem>Distributed File System</elem>
+-- <elem>Leasing</elem>
+-- <elem>Multi-credit operations</elem>
+-- </table>
+---
+
+author = "Paulino Calderon"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = {"safe", "discovery"}
+
+hostrule = function(host)
+  return smb.get_port(host) ~= nil
+end
+
+action = function(host,port)
+  local status, smbstate, overrides 
+  local output = stdnse.output_table()
+  overrides = {}
+
+  local smb2_dialects = {0x0202, 0x0210, 0x0300, 0x0302, 0x0311}
+
+  for i, dialect in pairs(smb2_dialects) do
+    -- we need a clean connection for each negotiate request
+    status, smbstate = smb.start(host)
+    if(status == false) then
+      return false, smbstate
```

Comment:
Don't return `false` from the `action` function. Either return nil (no output) or return an error message. I guess this was probably part of a function at one time.